Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -1874,6 +1879,59 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         }
     }
     
+    private func synthesizeArticleOnlyRootPage(articles: inout [DocumentationContext.SemanticResult<Article>], bundle: DocumentationBundle) throws {
+        let title = bundle.displayName
+        let markup = Document(
+            parsing: """
+                # \(title)
+                
+                @Metadata {
+                  @TechnologyRoot
+                }
+                """,
+            source: nil,
+            options: .parseBlockDirectives
+        )
+        
+        guard markup.childCount == 2,
+              let blockDirective = markup.child(at: 1) as? BlockDirective,
+              let metadata = Metadata(from: blockDirective, for: bundle, in: self)
+        else {
+            throw ContextError.unableToSynthesizeRootPage
+        }
```

Comment:
@d-ronnqvist Since `markup` is a constant and this is a programmer error would this not be better handled with an assertion?

Removing the guard statement also lets us get rid of the throwing function, which I think it makes sense here because there's nothing that could fail during runtime in this function.

Reply:
I'm a little bit hesitant to trusting the parsed content completely since it also contains the `# \(title)` which comes from the bundle display name and I don't know if there's any content that a developer could accidentally put in there that would break the markup and cause a hard to debug crash. 

That said, the markup is simple enough that I can manually construct it and avoid the parsing all together. I did this in https://github.com/apple/swift-docc/pull/778/commits/43ff6d4d46558f9d1e89820141ff8616b218cb87